src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -774,10 +740,26 @@ public static BigInteger DivRem(BigInteger dividend, BigInteger divisor, out Big
             if (trivialDivisor)
             {
                 uint rest;
-                uint[] bits = BigIntegerCalculator.Divide(dividend._bits, NumericsHelpers.Abs(divisor._sign), out rest);
 
-                remainder = dividend._sign < 0 ? -1 * rest : rest;
-                return new BigInteger(bits, bits, (dividend._sign < 0) ^ (divisor._sign < 0));
+                uint[]? bitsFromPool = null;
+                int size = dividend._bits.Length;
+                Span<uint> quotient = size <= BigIntegerCalculator.StackAllocThreshold ?
+                                  stackalloc uint[size]
+                                  : (bitsFromPool = ArrayPool<uint>.Shared.Rent(size)).AsSpan(0, size);
+
+                try
```

Comment:
As per the input from Jan/Stephen, we should remove these try/finally blocks and just not return the array when exceptions occur.